OvmfPkg/Library/QemuFwCfgS3Lib/QemuFwCfgS3PeiDxe.c

```diff
@@ -32,6 +33,13 @@ QemuFwCfgS3Enabled (
   UINTN                 FwCfgSize;
   UINT8                 SystemStates[6];
 
+ #ifdef MDE_CPU_X64

```

Comment:
The #ifdef should not be needed, there is a NULL version of the library for non-TDX builds.

Also I'd suggest to move the TDX check to OvmfPkg/PlatformPei/Platform.c where QemuFwCfgS3Enabled() is called from.

Reply:
There is a build error without #ifdef since the LoongArchVirt build with QemuFwCfgS3Lib .

QemuFwCfgS3Enabled() is called in OvmfPkg/PlatformPei/Platform.c and OvmfPkg/Library/AcpiPlatformLib/QemuFwCfgAcpi.c. 
To reduce the redundant logic code, move it in QemuFwCfgS3Enabled.

Reply:
> There is a build error without #ifdef since the LoongArchVirt build with QemuFwCfgS3Lib .

Probably due to the TDX NULL library missing there.

Reply:
> > There is a build error without #ifdef since the LoongArchVirt build with QemuFwCfgS3Lib .
> 
> Probably due to the TDX NULL library missing there.

TdIsEnabled() is built in BaseLib, but the NULL Version is built only for IA32, so there is the build error for LoongArchVirt.
There are two ways to resolve the build error:
1: add the #ifdef.
2: add the NULL Version for others (eg, LoongArch).
Which option do you prefer?

Reply:
I think it makes sense to use the NULL library not only for IA32 but all architectures which do not support TDX, so we do not need two different methods.  So I prefer (2).

Reply:
Updated with solution(2), and update the codebase to resolve the CI error.